src/Microsoft.SqlTools.ServiceLayer/Metadata/SmoScripterFactory.cs

```diff
@@ -0,0 +1,191 @@
+                DriWithNoCheck = false,
+                EnforceScriptingOptions = true,
+                ExtendedProperties = false,
+                FullTextCatalogs = false,
+                FullTextIndexes = false,
+                FullTextStopLists = false,
+                IncludeDatabaseContext = false,
+                IncludeDatabaseRoleMemberships = false,
+                IncludeFullTextCatalogRootPath = false,
+                IncludeHeaders = false,
+                IncludeIfNotExists = false,
+                IncludeScriptingParametersHeader = false,
+                Indexes = false,
+                LoginSid = false,
+                NoAssemblies = true,
+                NoCollation = true,
+                NoCommandTerminator = true,
+                NoExecuteAs = true,
+                NoFileGroup = true,
+                NoFileStream = true,
+                NoFileStreamColumn = true,
+                NoIdentities = true,
+                NoIndexPartitioningSchemes = true,
+                NoMailProfileAccounts = true,
+                NoMailProfilePrincipals = true,
+                NonClusteredIndexes = false,
+                NoTablePartitioningSchemes = true,
+                NoVardecimal = false,
+                NoViewColumns = false,
+                NoXmlNamespaces = false,
+                OptimizerData = false,
+                Permissions = false,
+                PrimaryObject = true,
+                SchemaQualify = true,
+                SchemaQualifyForeignKeysReferences = true,
+                ScriptBatchTerminator = false,
+                ScriptData = false,
+                ScriptDataCompression = false,
+                ScriptDrops = false,
+                ScriptForAlter = false,
+                ScriptForCreateDrop = false,
+                ScriptForCreateOrAlter = true,
+                ScriptOwner = false,
+                ScriptSchema = true,
+                ScriptXmlCompression = false,
+                SpatialIndexes = false,
+                Statistics = false,
+                TimestampToBinary = false,
+                ToFileOnly = false,
+                Triggers = false,
+                WithDependencies = false,
+                XmlIndexes = false                
+            };
+
+            var scripter = new Scripter(server);
+            scripter.Options = scriptingOptions;
+            var generatedScripts = await Task.Run(() => scripter.Script(urns));
+
+            var scripts = new List<string>();
+            foreach (var s in generatedScripts)
+            {
+                var script = s.Replace("\r", string.Empty).Replace("\n", string.Empty);
+                scripts.Add(script);
+            }
+
+            return scripts;
+        }
+
+        private static UrnCollection GetAllServerTableAndViewUrns(Server server)
+        {
+            UrnCollection urnCollection = new UrnCollection();
+
+            foreach (Database db in server.Databases)
```

Comment:
If anything goes wrong with querying the info for a database (such as not having access) then this whole thing will fail. Would suggest either scoping this to just the database the connection is for or at least putting all operations into try/catches so that it will at least get some of the server info back even if it can't get it all.